diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 01dbf2d..2913632 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -15,6 +15,8 @@ if (ARGP_INCLUDE_DIR)
     include_directories(${ARGP_INCLUDE_DIR})
 endif()
 
+add_definitions(-Wno-embedded-directive)
+
 if (UNIX AND NOT WIN32)
     add_executable(libssh_scp libssh_scp.c ${examples_SRCS})
     target_compile_options(libssh_scp PRIVATE ${DEFAULT_C_COMPILE_FLAGS})
--- libssh-0.10.3.orig/examples/sshnetcat.c	2023-01-20 10:12:35.851493458 +0000
+++ libssh-0.10.3/examples/sshnetcat.c	2023-01-20 10:12:46.888146303 +0000
@@ -238,7 +238,7 @@
 }
 
 void cleanup_pcap(void);
-void cleanup_pcap(){
+void cleanup_pcap(void){
 	ssh_pcap_file_free(pcap);
 	pcap=NULL;
 }
--- libssh-0.10.3.orig/examples/ssh_X11_client.c	2023-01-20 11:55:02.049530448 +0000
+++ libssh-0.10.3/examples/ssh_X11_client.c	2023-01-20 12:29:28.215151719 +0000
@@ -453,7 +453,7 @@
 
 
 static int
-x11_connect_display()
+x11_connect_display(void)
 {
 	int display_number;
 	const char *display = NULL;
